buildcheck/views/reviewer_validation.py

```diff
@@ -3,51 +3,72 @@
 from buildcheck.components.status_tag import status_tag
+from buildcheck.state.user_state import UserState
 from buildcheck.backend.supabase_client import supabase_client
 
+class AIValidationState(UserState):
+    violations: list[str] = []
+    guidelines: list[dict] = []
+    case_data: list[dict] = []
+    case_result: str = "No cases assigned"  # Default case result status
+    case_id: str = ""
+    listOfCases: list[str] = []
+    comments: dict = {}
 
-class GuidelineState(rx.State):
-    guideline: list[str]  # Explicitly type as list of list of strings
-    # I know that we should'nt use this table, but I'm only using it to get guidelines examples
-    # When we finalize the DB schema, I will replace this with the correct table
-    response = supabase_client.table("guidelines").select("*").execute()
-    guideline = [
-        [item.get("code"), item.get("title"), item.get("description"),item.get("category")]
-        for item in response.data
-    ]
-class SelectState(rx.State):
-    value: str = "444"
+    @rx.event
+    def load_data(self):
+        # Loads the case data for the current user from the database
+        try:
+            user_id_int = int(self.user_id)
+            response1 = supabase_client.table("cases").select("*").eq("reviewer_id", user_id_int).execute()
+            self.case_id = str(response1.data[0]["id"])
+            self.case_data = response1.data
+            self.listOfCases = [str(case["id"]) for case in self.case_data]
+            response2 = supabase_client.table("violations").select("guideline_code").eq("case_id", self.case_id).execute()
+            self.violations = [row["guideline_code"] for row in response2.data]
+            response3 = supabase_client.table("guidelines").select("*").execute() 
+            self.guidelines = response3.data
+        except Exception as e:
+            print("Exception in load_data:", e)
 
     @rx.event
-    def change_bluePrint(self, value: str):
-        self.value = value
+    def change_case(self, value: str):
+        self.case_id = value
 
-def show_guideline(guideline: list):
-    # Trigger guideline loading if needed (handled by State in Reflex)
-    return rx.table.row(
-        rx.table.cell(guideline[0]),
-        rx.table.cell(guideline[1]),
-        rx.table.cell(guideline[2]),
-        rx.table.cell(status_tag("approved")),  # Assuming status is always "approved" for this example
-        rx.table.cell(guideline[3]),
-        rx.table.cell(rx.button("Delete", color_scheme="red")),
-    )
-class FormInputState(rx.State):
-    form_data: dict = {}
+    @rx.event
+    def handle_comments(self, form_data: dict):
+        # Store the comment locally
+        self.comments = form_data
+        # Update the comments column in the cases table for the selected case_id
+        try:
+            comment_text = form_data.get("comment", "")
+            supabase_client.table("cases").update({"comments": comment_text}).eq("id", self.case_id).execute()
```

Comment:
This column should be `reviewer_comment`.

Reply:
thats fine but why cant we just use `reviewer_comment` to hold a single comment for now